ssc/libssc.c

```diff
@@ -1108,14 +1585,157 @@ static PetscErrorCode PCSetUpOnBlocks_PATCH(PC pc)
   PetscFunctionReturn(0);
 }
 
-#undef __FUNCT__
-#define __FUNCT__ "PCSetFromOptions_PATCH"
+PETSC_EXTERN PetscErrorCode PCPatchConstruct_Star(void *vpatch, DM dm, PetscInt entity, PetscHashI ht)
+{
+    PetscErrorCode ierr;
+    PetscInt       starSize, closureSize, fclosureSize;
+    PetscInt      *star = NULL, *closure = NULL, *fclosure = NULL;
+    PetscInt       vStart, vEnd;
+    PetscInt       fStart, fEnd;
+    PetscInt       cStart, cEnd;
+
+    PetscFunctionBegin;
+    PetscHashIClear(ht);
+
+    /* Find out what the facets and vertices are */
+    ierr = DMPlexGetHeightStratum(dm, 1, &fStart, &fEnd); CHKERRQ(ierr);
+    ierr = DMPlexGetDepthStratum(dm,  0, &vStart, &vEnd); CHKERRQ(ierr);
+    ierr = DMPlexGetHeightStratum(dm, 0, &cStart, &cEnd); CHKERRQ(ierr);
+
+    /* To start with, add the entity we care about */
+    PetscHashIAdd(ht, entity, 0);
+
+    /* Loop over all the cells that this entity connects to */
+    ierr = DMPlexGetTransitiveClosure(dm, entity, PETSC_FALSE, &starSize, &star); CHKERRQ(ierr);
+    for ( PetscInt si = 0; si < starSize; si++ ) {
+        PetscInt cell = star[2*si];
+        if ( cell < cStart || cell >= cEnd) continue;
```

Comment:
I think this is a very complicated way of copying the star of the vertex into a hash table.

Reply:
Fixed.